image-registry/remove-registry-baremetal.md

```diff
@@ -0,0 +1,190 @@
+  cleanly if the registry is not installed.
+
+## Proposal
+
+This is where we get down to the nitty gritty of what the proposal actually is.
+
+### User Story
+
+As an OpenShift cluster admin/installer
+I want the registry to be `Removed` on RHV and k8s-native infra installations
+So that I can provision other storage providers
+And later enable the image registry with the desired storage
+
+### Implementation Details/Notes/Constraints
+
+- When the image registry operator bootstraps itself, it will inspect the
+  cluster infrastructure platform status.
+- The registry will bootstrap itself as `Removed` on the following platform types:
+  - `BareMetal` (i.e. baremetal platforms that support IPI install)
+  - `oVirt` (RHV)
+  - `VSphere`
+  - `None` (true UPI baremetal installs)
+- The registry will bootstrap itself as `Managed` with `EmptyDir` storage on
+  the following platform types:
+  - `Libvirt`
+  - `Unknown` platforms
+- The registry will continue to bootstrap itself as `Managed` with specific
+  storage on the following platforms:
+  - `AWS` (Amazon S3)
+  - `Azure` (Azure blob storage)
+  - `GCP` (Google Cloud Storage)
+  - `OpenStack` (Swift)
+- Any platform that wishes to set up the registry must perform a day 2 action
+  that does the following:
+  - Provisions suitable storage, typically a PV that supports the `RWX` access
+    mode.
+  - Updates the image registry operator's management state to `Managed`.
+- Telemeter should be aware if a cluster's registry has been removed:
+  - Reason codes in the ClusterOperator conditions (`Available` in particular).
+  - Prometheus alerts warning the registry has been removed.
+
+### Risks and Mitigations
+
+#### Image registry is never installed
+
+Several components will likely report errors if the registry is `Removed` and
+an admin does nothing to set up the registry. Potential candidates that will
+break:
+
+1. Builds which push to an imagestream tag will error at runtime.
+2. Samples operator imagestreams which import from `registry.redhat.io` will
+   require a pull secret to be usable.
+3. Imagestreams imported with `--reference-policy=local` will not reference the
+   internal registry as expected. `BuildConfig` and `DeploymentConfig` objects
+   which do their own resolution of `ImageStreamTag` will break.
+
+Mitigations:
+
+1. Existing error messages from affected components.
+2. Image registry's ClusterOperator will provide `Reason="Removed"` if the
+   registry remains removed.
+3. If removed, the image registry operator should fire an alert warning that:
+   1. The registry has been removed.
+   2. The following components may not work as expected:
+      1. `ImageStreamTags`
+      2. `BuildConfigs` which reference `ImageStreamTags`
+      3. `DeploymentConfigs` which reference `ImageStreamTags`
+   3. Recommends admins configure storage and update the operator config to
+   the `Managed` state (via a doc link)
```

Comment:
Do we have a way of linking to a doc page that is always pointing to the `current`? I am afraid this link may end up pointing to a "deprecated" documentation later down the line.

Reply:
unfortunately I do not believe we do (we used to have "latest" docs for OKD, but for OCP it's always version specific).  I'd be wary of pointing to latest anyway since that would mean someone who hadn't upgraded would be getting directed to newer docs that might not be appropriate.

it might be best to just punt on having a doc link here, unless @bmcelvee has some ideas.

Reply:
I'd say we don't have "deprecated" documentation for supported versions. If we do add an important notice to our documentation, we usually backport it to all relevant versions. But despite having the same codebase, the links for OKD and OCP should be different (a compile-time pattern for URLs?). And it's additional maintenance burden to check if these links are valid for a new release.